clients/admin-ui/src/features/privacy-requests/DenyPrivacyRequestModal.tsx

```diff
@@ -6,35 +6,33 @@ import {
   ModalFooter,
   ModalHeader,
   ModalOverlay,
-  Textarea,
 } from "@fidesui/react";
+import { CustomTextArea } from "common/form/inputs";
+import { Form, Formik } from "formik";
 import React from "react";
+import * as Yup from "yup";
 
 type DenyModalProps = {
   isOpen: boolean;
-  isLoading: boolean;
   handleMenuClose: () => void;
   handleDenyRequest: (reason: string) => Promise<any>;
-  denialReason: string;
-  onChange: (e: any) => void;
 };
 
-const closeModal = (
+const handleSubmit = (
```

Comment:
nit: you could put this func inside of `DenyPrivacyRequestModal` so that you don't have to pass in those functions. then your `handleSubmit` can look like

```typescript
type FormValues = typeof initialValues;

const handleSubmit = (values: FormValues, formikHelpers: FormikHelpers<FormValues>) => {
  const { setSubmitting } = formikHelpers;
  handleDenyRequest(values.denialReason).then(() => {
    setSubmitting(false);
    handleMenuClose();
  });
};
```

I find this helpful for typing reasons since then `handleSubmit` knows exactly what it has access to.

Reply:
incidentally, I totally thought formik would handle setting `isSubmitting=false` but it turns out it only does this if the `onSubmit` is async! 🤯 https://formik.org/docs/api/formik#onsubmit-values-values-formikbag-formikbag--void--promiseany